### DISCUSSION TITLE
[Feature] Change date and time relatively / by offset

### DISCUSSION THREAD

**Discussion by flesser** [immich-app/immich#9430] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Sometimes, I have a bunch of images where the date is off by e.g. an hour (daylight saving time issue) or even several years (camera date fell back to manufacturing year).
I would like to be able to correct those image dates by adding/subtracting a certain offset for each images date (similar to [jhead}(https://linux.die.net/man/1/jhead) with its `-ta` / `-da` flags).

Currently, if you have more than one image selected and use "adjust date/time", all images are set to the same date/time, loosing their relative time information.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by okwes** [comment]
I had this issue as I uploaded over 1,000 photos with wrong exif times... as a temporary stop gap I wrote a simple python script that will call the API and shift the time for each item in an album. I wrote it all very quickly so no guarantees, but maybe it will be helpful to someone:)

[https://gist.github.com/okwes/d00c6e064a5ba7731ff31db853489dcc](https://gist.github.com/okwes/d00c6e064a5ba7731ff31db853489dcc)

**Comment by Qhilm** [comment]
This seems to be an option too (not tested yet):

https://github.com/varun-raj/immich-power-tools

**Comment by Toastie** [comment]
Ran into this exact issue today, had to reupload a bunch of images as I mistakenly set them all to the same time when adjusting the timezone.

So adding this feature would be much appreciated.

**Comment by mkuehne707** [comment]
I put together a PR that would implement this:
https://github.com/immich-app/immich/pull/17717
